api-server/src/server/boot/challenge.js

```diff
@@ -273,63 +272,60 @@ export function isValidChallengeCompletion(req, res, next) {
   return next();
 }
 
-export function modernChallengeCompleted(req, res, next) {
+export async function modernChallengeCompleted(req, res, next) {
   const user = req.user;
-  return user
-    .getCompletedChallenges$()
-    .flatMap(() => {
-      const completedDate = Date.now();
-      const { id, files, challengeType } = req.body;
-
-      const completedChallenge = {
-        id,
-        files,
-        completedDate
-      };
-
-      // if multifile cert project
-      if (challengeType === 14) {
-        completedChallenge.isManuallyApproved = false;
-        user.needsModeration = true;
-      }
 
-      // We only need to know the challenge type if it's a project. If it's a
-      // step or normal challenge we can avoid storing in the database.
-      if (
-        jsCertProjectIds.includes(id) ||
-        multifileCertProjectIds.includes(id)
-      ) {
-        completedChallenge.challengeType = challengeType;
-      }
+  try {
+    // This is an ugly way to update `user.completedChallenges`
+    await user.getCompletedChallenges$().toPromise();
+  } catch (e) {
+    return next(e);
+  }
```

Comment:
What does this actually do? I couldn't see anything obviously broken when I commented out this section.

Reply:
I know it runs that code, I just don't understand what purpose it serves.

To put it another way: under what circumstances is `user.completedChallenges` in need of an update?

Reply:
It's a little out of scope, so if you'd rather we address it later, that's fine with me.

Reply:
If you are asking why/how `completedChallenges` got updated between the initial fetch of the `user` object and this line of code, then I do not know:

![image](https://user-images.githubusercontent.com/51722130/220977672-4faf4750-e81b-4ef3-9f66-fc17756b5c8e.png)

Reply:
![image](https://user-images.githubusercontent.com/51722130/220980400-bd929cbe-763c-4669-84f5-d0af8e7c20f6.png)

Reply:
Fair. Let's handle that in another PR.